tests/benchmark/challenges/test_rosenbrock.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+from unittest import TestCase
+from unittest.mock import call, patch
+
+from btb.benchmark.challenges.rosenbrock import Rosenbrock
+
+
+class TestRosenbrock(TestCase):
+
+    def test___init__default(self):
+        # run
+        rosenbrock = Rosenbrock()
+
+        # assert
+        assert rosenbrock.a == 1
+        assert rosenbrock.b == 100
+        assert rosenbrock.min_x == -50
+        assert rosenbrock.max_x == 50
+        assert rosenbrock.min_y == -50
+        assert rosenbrock.max_y == 50
+
+    def test___init__custom(self):
+        # run
+        rosenbrock = Rosenbrock(a=2, b=3, min_x=4, max_x=5, min_y=6, max_y=7)
+
+        # assert
+        assert rosenbrock.a == 2
+        assert rosenbrock.b == 3
+        assert rosenbrock.min_x == 4
+        assert rosenbrock.max_x == 5
+        assert rosenbrock.min_y == 6
+        assert rosenbrock.max_y == 7
+
+    @patch('btb.benchmark.challenges.rosenbrock.IntHyperParam')
+    @patch('btb.benchmark.challenges.rosenbrock.Tunable')
+    def test_get_tunable(self, mock_tunable, mock_inthyperparam):
+        # setup
+        mock_inthyperparam.side_effect = [1, 2]
+        mock_tunable.return_value = 'tunable'
+
+        # run
+        result = Rosenbrock().get_tunable()
+
+        # assert
+        assert result == 'tunable'
+        assert mock_inthyperparam.call_args_list == [call(min=-50, max=50), call(min=-50, max=50)]
```

Comment:
is it not better to use `mock_inthyperparam.assert_has_calls`? See https://docs.python.org/3.5/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls. Unless there is some py2 compatibility issue? If so it is nice and clean to keep all py2 compatibility in one location, such as `btb.compat`, and defining a shim there.